examples/restore_login.py

```diff
@@ -1,71 +1,90 @@
+#!/usr/bin/env python3
+
 import asyncio
 import json
 import os
 import sys
+import getpass
 
 from nio import AsyncClient, LoginResponse
 
 CONFIG_FILE = "restore_login_config.json"
 
 # Check out main() below to see how it's done.
 
-def write_details_to_disk(resp: LoginResponse) -> None:
+
+def write_details_to_disk(resp: LoginResponse, home_server) -> None:
     """Writes the required login details to disk so we can log in later without
     using a password.
 
     Arguments:
         resp {LoginResponse} -- the successful client login response.
+        home_server -- the name of the homeserver, e.g. "https://matrix.example.org"
     """
     # open the config file in write-mode
     with open(CONFIG_FILE, "w") as f:
         # write the login details to disk
         json.dump(
             {
-            "access_token": resp.access_token,
-            "device_id": resp.device_id,
-            "user_id": resp.user_id
+                "home_server": home_server,  # e.g. "https://matrix.example.org"
+                "access_token": resp.access_token,  # long cryptographic access token
+                "device_id": resp.device_id,  # device ID, 10 uppercase letters
+                "user_id": resp.user_id  # e.g. "@user:example.org"
```

Comment:
I don't think you need to provide any comments here on what's what.

Reply:
I think some new comers might find this helpful.
Since it does not add a single line of code, can we leave it in?

Reply:
You're right that it doesn't add lines, but it does add bulk. It's extra noise to read. I'm not sure that anyone is going to get confused (at this point in the code) about what a homeserver, device ID, access token, or user ID are.